prism/src/main/java/org/apache/falcon/resource/AbstractEntityManager.java

```diff
@@ -354,6 +357,12 @@ protected APIResult update(Entity newEntity, String type, String entityName, Boo
             }
 
             configStore.update(entityType, newEntity);
+            // for Data Sources, check always if dependant feeds are already upgraded and upgrade accordingly
+            if (entityType.equals(EntityType.DATASOURCE)) {
```

Comment:
Minor suggestion feel free to ignore : We have different tasks to be performed for different entityType. May be having a switch on entityType and moving corresponding update commands to the right case will make it more readable.